### DISCUSSION TITLE
How do I replace control?

### DISCUSSION THREAD

**Discussion by adammo94** [JedWatson/react-select#4998] [Q&A]
https://codesandbox.io/s/96w3y?module=/example.tsx Here is an example for custom control by this library creators.
The problem is control is not being replaced. Instead it just adds a wrapper around the control.
```
<div style={controlStyles}>
  <p>Custom Control</p>
  <components.Control {...props} />
</div>
```
What's the correct way to replace control element itself with my own component but without losing props?

**Comment by Methuselah96** [comment]
If you need to just add props to the Control `div` I would suggest using the `innerProps` with something like this (untested):
```jsx
const { innerProps, ...rest } = props;
return <components.Control { ...rest, innerProps: { ...innerProps, someDivProp: someValue }} />;
```

Otherwise you could just copy-paste the `Control` component from [the code](https://github.com/JedWatson/react-select/blob/master/packages/react-select/src/components/Control.tsx#L82-L98) and replace it with what you want instead.
